### PR TITLE
created the model and migration file of the hotels table

### PR DESCRIPTION
**Pull request by anichukwuebuka65**

### CODE REVIEW

File: database/migrations/2023_09_14_213354_create_hotels_table.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('hotels', function (Blueprint $table) {
+            $table->integer('user_id');
```
Comment (davydocsurg) [inline database/migrations/2023_09_14_213354_create_hotels_table.php:15]: @anichukwuebuka65 Please review the tables that already exist and make any necessary corrections.

**Review by davydocsurg — CHANGES_REQUESTED** [review]

File: database/migrations/2023_09_14_213354_create_hotels_table.php (hunk 1 above)
Comment (anichukwuebuka65) [reply database/migrations/2023_09_14_213354_create_hotels_table.php]: I have made the changes. The migrations now run successfully.

File: database/migrations/2023_09_16_080713_create_hotels_table.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('hotels', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
+            $table->foreign('user_id')->references('id')->on('users');
```
Comment (davydocsurg) [inline database/migrations/2023_09_16_080713_create_hotels_table.php:17]: Let's use this instead: `$table->foreignId('user_id')->constrained('users')->onDelete('cascade')->onUpdate('cascade');`

File: database/migrations/2023_09_16_080713_create_hotels_table.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            $table->char('hotel_name');
```
Comment (davydocsurg) [inline database/migrations/2023_09_16_080713_create_hotels_table.php:18]: Please, replace `$table->char()` with `$table->string()` and remove the limit - 255

File: database/migrations/2023_09_16_080713_create_hotels_table.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            $table->char('description', 255);
+            $table->integer('location');
+            $table->integer('rating');
+            $table->integer('price_range');
+            $table->char('amenities', 255);
+            $table->date('check_in_time');
```
Comment (davydocsurg) [inline database/migrations/2023_09_16_080713_create_hotels_table.php:24]: replace `$table->date()` with `$table->dateTime()`

**Review by davydocsurg — CHANGES_REQUESTED** [review]

File: database/migrations/2023_09_16_080713_create_hotels_table.php (hunk 4 above)
Comment (anichukwuebuka65) [reply database/migrations/2023_09_16_080713_create_hotels_table.php]: I have applied the changes

**Review by davydocsurg — APPROVED** [review]

**Pull request merged**